UserStudentApp/models.py

```diff
@@ -0,0 +1,98 @@
+from django.db import models    
+from django.conf import settings
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+from rest_framework.authtoken.models import Token
+from django.contrib.auth.models import AbstractBaseUser, BaseUserManager, PermissionsMixin
+
+class CustomManager(BaseUserManager):
+    def create_user(self, email, username, password=None):
+        if not email:
+            raise ValueError('User must have an Email')
+        if not username:
+            raise ValueError('User must have an Username')
+        user = self.model(
+            email = self.normalize_email(email),
+            username = username,
+        )
+        user.set_password(password)
+        user.save(using = self._db)
+        return user
+
+    def create_superuser(self,email,username,password):
+        user = self.create_user(
+            email    = self.normalize_email(email),
+            username = username,
+            password = password,
+        )
+        user.is_admin     = True
+        user.is_active    = True
+        user.is_superuser = True
+        user.save(using = self._db)
+        return user
+
+def get_picture_path(self, filename):
+    return f'profile_pics/{self.pk}/{"profile_img.jpg"}'
+
+def get_other_pictures(self, filename):
+    return f'other_pics/{self.pk}/{"pic_name.jpg"}'
```

Comment:
These functions will create a 2 dirs 
1 is profile_pics where all the profile pics will be uploaded and inside this dir every user has it profile pic inside his ID which is a cleaner way in my views and "profile_img.jpg" will change the filename so that if 2 similar named pics were uploaded it will not override the previous image.
so they are very similar but one for profile_pics and other one is for other_pics.